Sources/HTTP/HTTPServer.swift

```diff
@@ -0,0 +1,46 @@
+// This source file is part of the Swift.org Server APIs open source project
+//
+// Copyright (c) 2017 Swift Server API project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+//
+
+/// Definition of an HTTP server.
+public protocol HTTPServer : class {
+
+    /// Start the HTTP server on the given `port`, using `handler` to process incoming requests
+    func start(port: Int, handler: @escaping HTTPRequestHandler) throws
+
+    /// Stop the server
+    func stop()
+
+    /// The port the server is listening on
+    var port: Int { get }
+
+    /// The number of current connections
+    var connectionCount: Int { get }
+}
+
+/// A basic HTTP server. Currently this is implemented using the BlueSocket
+/// abstraction, but the intention is to remove this dependency and reimplement
+/// the class using transport APIs provided by the Server APIs working group.
+public class SimpleHTTPServer {
```

Comment:
I am not sure to understand the purpose of this public class. If it is something that devs are supposed to use, wouldn't it be better to call the protocol `HTTPServerType` and this class just `HTTPServer`?